src/Parser/HParser.y

```diff
@@ -0,0 +1,299 @@
+         | '+'                                { withPos1 $1 $ varE "+" }
+         | '-'                                { withPos1 $1 $ varE "-" }
+         | '^'                                { withPos1 $1 $ varE "^" }
+         | '<'                                { withPos1 $1 $ varE "<" }
+         | '>'                                { withPos1 $1 $ varE ">" }
+         | '<='                               { withPos1 $1 $ varE "<=" }
+         | '>='                               { withPos1 $1 $ varE ">=" }
+         | '<>'                               { withPos1 $1 $ varE "<>" }
+         | '=='                               { withPos1 $1 $ varE "==" }
+         | '&&'                               { withPos1 $1 $ varE "&&" }
+         | '||'                               { withPos1 $1 $ varE "||" }
+
+Var :: { Located (Var Parsed) }
+    : ident { lPos1 $1 $ Name (getIdent $1) }
+
+Con :: { Located (Var Parsed) }
+    : conid { lPos1 $1 $ Name (getIdent $1) }
+
+TyVar :: { Var Parsed }
+      : tyvar { Name (getIdent $1) }
+
+BindGroup :: { [(Var Parsed, Expr Parsed, Ann Parsed)] }
+          : Binding                           { [$1] }
+          | BindGroup and Binding             { $3 : $1 }
+
+Binding :: { (Var Parsed, Expr Parsed, Ann Parsed) }
+        : Var ListE(ArgP) '=' Expr            { (getL $1, foldr (\x y -> withPos2 x $4 (Fun x y)) $4 $2, withPos2 $1 $4 id) }
+
+List(p, s)
+    : {- Empty -}       { [] }
+    | List1(p, s)       { $1 }
+
+List1(p, s)
+     : p                { [$1] }
+     | p s List1(p, s)  { $1 : $3 }
+
+ListE(p)
+     : {- Empty -}      { [] }
+     | ListE1(p)        { $1 }
+
+ListE1(p)
+     : p                { [$1] }
+     | p ListE1(p)      { $1 : $2 }
+
+Rows(p, q)
+   : {- Empty -}             { [] }
+   | ident p q ',' Rows(p,q) { (getIdent $1, $3) : $5 }
+   | ident p q               { [(getIdent $1, $3)] }
+
+Lit :: { Located Lit }
+    : int                  { lPos1 $1 $ LiInt (getInt $1) }
+    | string               { lPos1 $1 $ LiStr (getString $1) }
+    | true                 { lPos1 $1 $ LiBool True }
+    | false                { lPos1 $1 $ LiBool False }
+
+Pattern :: { Pattern Parsed }
+        : ArgP           { $1 }
+        | Con Pattern    { withPos2 $1 $2 $ Destructure (getL $1) (Just $2) }
+
+ArgP :: { Pattern Parsed }
+     : Var                        { withPos1 $1 $ Capture (getL $1) }
+     | '_'                        { withPos1 $1 $ Wildcard }
+     | Con                        { withPos1 $1 $ Destructure (getL $1) Nothing }
+     | '{' Rows('=',Pattern) '}'  { withPos2 $1 $3 $ PRecord $2 }
+     | '(' List(Pattern, ',') ')' { withPos2 $1 $3 $ tuplePattern $2 }
+     | ArgP ':' Type              { withPos2 $1 $2 $ PType $1 $3 }
+
+Arm :: { (Pattern Parsed, Expr Parsed) }
+    : '|' Pattern '->' Expr       { ($2, $4) }
+
+
+Type :: { Type Parsed }
+     : TypeProd                                   { $1 }
+     | TypeProd '->' Type                         { TyArr $1 $3 }
```

Comment:
Does this mean `a * b -> c` is `(a * b) -> c` or `a * (b -> c)`?

Reply:
`(a * b) -> c` IIRC. I'll double check though.